src/computability/language.lean

```diff
@@ -151,4 +159,147 @@ begin
     finish }
 end
 
+lemma le_iff (l m : language α) : l ≤ m ↔ l + m = m := sup_eq_right.symm
+
+lemma le_mul_congr {l₁ l₂ m₁ m₂ : language α} : l₁ ≤ m₁ → l₂ ≤ m₂ → l₁ * l₂ ≤ m₁ * m₂ :=
+begin
+  intros h₁ h₂ x hx,
+  simp only [mul_def, exists_and_distrib_left, set.mem_image2, set.image_prod] at hx ⊢,
+  tauto
+end
+
+lemma le_add_congr {l₁ l₂ m₁ m₂ : language α} : l₁ ≤ m₁ → l₂ ≤ m₂ → l₁ + l₂ ≤ m₁ + m₂ :=
+begin
+  intros h₁ h₂ x hx,
+  simp only [add_def, set.mem_union_eq] at hx ⊢,
+  tauto
+end
+
+lemma supr_mul {ι : Type v} (l : ι → language α) (m : language α) :
+  (⨆ i, l i) * m = ⨆ i, l i * m :=
+begin
+  ext x,
+  simp only [mem_mul, set.mem_Union, exists_and_distrib_left],
+  tauto
+end
+
+lemma mul_supr {ι : Type v} (l : ι → language α) (m : language α) :
+  m * (⨆ i, l i) = ⨆ i, m * l i :=
+begin
+  ext x,
+  simp only [mem_mul, set.mem_Union, exists_and_distrib_left],
+  tauto
+end
+
+lemma supr_add {ι : Type v} [nonempty ι] (l : ι → language α) (m : language α) :
+  (⨆ i, l i) + m = ⨆ i, l i + m :=
+begin
+  ext x,
+  simp only [mem_add, set.mem_Union],
+  split,
+  { rintro (⟨ i, h ⟩ | h),
+    { use i,
+      exact or.inl h },
+    { finish } },
+  { finish }
+end
```

Comment:
It looks like this lemma is missing from `complete_lattice` - I'd expect a `supr_sup` lemma to match the existing [`infi_inf`](https://leanprover-community.github.io/mathlib_docs/order/complete_lattice.html#infi_inf) lemma (assuming that the lemma is indeed generally true)

Reply:
Although I would guess you can use something like this here:
```suggestion
@infi_inf (order_dual $ language) _ _ _ l m
```